server/map_rpc_server.go

```diff
@@ -49,11 +49,85 @@ func (t *TrillianMapServer) IsHealthy() error {
 	return t.registry.MapStorage.CheckDatabaseAccessible(context.Background())
 }
 
+// Init creates the initial revision 0 SignedMapHead, if one doesn't already exist.
+func (t *TrillianMapServer) Init(ctx context.Context, mapID int64) error {
+	tree, hasher, err := t.getTreeAndHasher(ctx, mapID, false /* readonly */)
+	if err != nil {
+		return err
+	}
+	ctx = trees.NewContext(ctx, tree)
+
+	tx, err := t.registry.MapStorage.BeginForTree(ctx, mapID)
+	if err != nil && err != storage.ErrMapNeedsInit {
```

Comment:
does `tx.Close()` need to be called when `err == storage.ErrMapNeedsInit` ?

Reply:
It's there mainly for the `err==nil` case.  For the `ErrMapNeedsInit` case it's a no-op when you get as far as the Commit but is useful for the intervening `return err`s.